File: 3042-count-prefix-and-suffix-pairs-i/3042-count-prefix-and-suffix-pairs-i.py
class Solution:
    def countPrefixSuffixPairs(self, words: List[str]) -> int:
        def ipas(str1, str2):
            if str2.startswith(str1) and str2.endswith(str1):
                return True
                
        ans = 0
        
        for i in range(len(words)):
            for j in range(i + 1, len(words)):
                if ipas(words[i], words[j]):
                    ans += 1
            
        return ans

File: 3042-count-prefix-and-suffix-pairs-i/3042-count-prefix-and-suffix-pairs-i.rs
impl Solution {
    pub fn count_prefix_suffix_pairs(words: Vec<String>) -> i32 {
        let mut ans = 0;

        fn check(str1: &str, str2: &str) -> bool {
            str2.starts_with(str1) && str2.ends_with(str1)
        }

        for i in 0..words.len() {
            for j in i + 1..words.len() {
                if check(&words[i], &words[j]) {
                    ans += 1;
                }
            }
        }

        ans
    }
}